### ISSUE TITLE
LLaMA sentiment example doesn't work

### ISSUE THREAD

**Issue by nikebless** [CarperAI/trlx#476]
### 🐛 Describe the bug

I am running into an issue when trying the TRLX [sentiment example with LLaMA](https://github.com/CarperAI/trlx/blob/main/examples/ppo_sentiments_llama.py).

The example uses [decapoda-research/llama-7b-hf](https://huggingface.co/decapoda-research/llama-7b-hf) with this [config](https://huggingface.co/decapoda-research/llama-7b-hf/blob/main/config.json), which expects `transformers` to refer to LLaMA as `LLaMAForCausalLM`. However, the `transformers` library (for over a month now) [refers](https://github.com/huggingface/transformers/blob/5d02e6bd2079a3691c5ef7a1c888abe6a16d854b/src/transformers/models/auto/modeling_auto.py#L382) to it as  `LlamaForCausalLM` (different capitalization). TRLX was updated to support the new naming (it even asserts the new naming [here](https://github.com/CarperAI/trlx/blob/9fdd0d757e8f7a3d48e7edb060ddb7517da13d2d/trlx/models/modeling_ppo.py#L1270). However, the example above doesn’t seem to have been updated and it crashes because of the naming mismatch.

It seems there’re several solutions:
1. Use an older, nightly version of transformers that uses the old naming (or make [your own fork](https://github.com/mbehm/transformers/commit/960e1f63b92ae05f0752e24247dc258a23e84ca4#diff-e4043b44ca10871d228d2c1f5a65af4612bb9cd9b95590917475ba573ca5fc91))
1. Don’t use Decapoda LLaMA, use your local LLaMA with a custom config
1. Use the weights from Decapoda LLaMA but replace their config with a custom config

#3 seems like the best solution for updating a public example.
* I was able to do that by downloading the Decapoda model from HF [using git](https://huggingface.co/docs/hub/models-downloading#using-git), changing the `config.json` in the downloaded folder, and then using the absolute path to that folder instead of the huggingface stub in the example code. 
* However, that's not a very friendly process for newcomers. I am considering diving deeper into TRLX model loading code and modifying the model config in the example code before it actually gets loaded. I am somewhat skeptical it's possible without touching TRLX at all. Before I attempt this, maybe someone has a better solution?

P.S. As discovered by @meg-tong, the Decapoda model is full of PRs trying to update the config, with the oldest one being 2 months old — so I wouldn't hope they'll fix it themselves 🙃 
P.S.S. The `tokenizer_config.json` also has to be updated in a similar way (from `LLamaTokenizer` to `LlamaTokenizer`)

### Which trlX version are you using?

0.6.0

### Additional system and package information

transformers==4.28.0, Python==3.10.10

**Comment by maxreciprocate** [comment]
This should have been resolved by updating transformers where this inconsistency was removed, see https://github.com/huggingface/transformers/issues/22222

**Comment by mbalesni** [comment]
IIRC, the transformers update you’re referring to is what caused the inconsistency. The Decapoda config referenced the names in an intermediate version of transformers, which worked before a final PR renamed things, breaking the Decapoda config. So the current transformers version is NOT a solution to this.

A solution would involve Decopoda updating their config, or TRLX using a different model for this example.

**Comment by maxreciprocate** [comment]
Correct, TRLX now uses a different model for this example: https://github.com/CarperAI/trlx/blob/a7eb066e518e860fa26a5801ca829a8c6ed1c044/examples/ppo_sentiments_llama.py#L42
